src/modules/configurator/services/API_configurations.ts

```diff
@@ -111,4 +111,29 @@ export const fetchAllPropertyImagesByVariant = async ({modelName, name, variant}
     } catch (err: any) {
         throw new Error(err);
     }
+}
+
+export const fetchUserConfiguration = async (userId: string, configurationId: string) => {
+    try {
+        const configRef = doc(db, `users/${userId}/configurations/${configurationId}`);
+        const response = await getDoc(configRef);
+        
+        const responseData = response.data();
+
+        if (!responseData) throw new Error("Data could not be found.");
+        const userConfig: UserCarConfiguration = {
+            model: responseData.model,
+            modelId: responseData.model_id,
+            productionYear: responseData.production_year,
+            color: responseData.color,
+            wheels: responseData.wheel_variant,
+            interior_variants: responseData.interior_variants,
+            creationDate: responseData.creation_date,
+            totalPrice: responseData.total_price
+        }
+
+        return userConfig;
```

Comment:
You could also return this object and set the return type of the function to be UserCarConfiguration or undefined.

Reply:
Added return types for all api calls.